great_expectations/datasource/fluent/interfaces.py

```diff
@@ -1302,88 +1287,40 @@ def compute_metrics(self, metrics: Metric | list[Metric]) -> MetricResult | list
-        }
-        return raw_metrics, raw_metrics_errors
 
-    @classmethod
-    def _parse_raw_metric_to_metric_result_value(
-        cls, metric_name: str, raw_metric: RawMetric
-    ) -> MetricValue:
-        # "condition" metrics return the domain and value kwargs
-        # we just want the value, which is the first item in the tuple
-        if metric_name.endswith(MetricNameSuffix.CONDITION) and isinstance(raw_metric.value, tuple):
-            value = raw_metric.value[0]
-        else:
-            value = raw_metric.value
-        return value
-
-    @classmethod
-    def metrics_calculator_result_to_metric_result(
-        cls,
-        requested_metric_names: list[str],
-        metrics_calculator_result: tuple[_MetricsDict, _AbortedMetricsInfoDict],
-    ) -> MetricResult | list[MetricResult]:
-        """Converts the result of MetricsCalculator.compute_metrics() into
-        a MetricResult or list[MetricResult]. Only the metrics that were
-        requested in the call to MetricsCalculator.compute_metrics()
-        will be returned (dependency metrics are excluded)."""
-        raw_metrics, raw_metrics_errors = cls._metrics_calculator_result_to_metrics_hashmaps(
-            metrics_calculator_result
-        )
-        metric_results = []
-        for metric_name in requested_metric_names:
-            if metric_name in raw_metrics:
-                MetricType = MetaMetric.get_registered_metric_class_from_metric_name(metric_name)
+        results = []
+        for metric in metrics:
+            metric_id_for_batch = metric.metric_id_for_batch(batch_id=self.id)
+            if metric_id_for_batch in metrics_calculator_results:
+                MetricType = MetaMetric.get_registered_metric_class_from_metric_name(metric.name)
                 MetricResultType = MetricType.get_metric_result_type()
-                value = cls._parse_raw_metric_to_metric_result_value(
-                    metric_name, raw_metrics[metric_name]
-                )
-                metric_results.append(
-                    MetricResultType(
-                        id=raw_metrics[metric_name].metric_configuration_id,
-                        value=value,
-                    )
-                )
-            elif metric_name in raw_metrics_errors:
-                metric_results.append(
-                    MetricErrorResult(
-                        id=raw_metrics_errors[metric_name].metric_configuration_id,
-                        value=raw_metrics_errors[metric_name].value,
+                value = metrics_calculator_results[metric_id_for_batch]
+                results.append(MetricResultType(id=metric_id_for_batch, value=value))
+            elif metric_id_for_batch in metrics_calculator_errors:
+                value = metrics_calculator_errors[metric_id_for_batch]
+                results.append(MetricErrorResult(id=metric_id_for_batch, value=value))
+            else:
```

Comment:
Added an `else` to enforce the one output per input. I saw this fail when the metric named changed for the spark mean failure (eg bug in gx).